fiftyone/core/odm/document.py

```diff
@@ -644,9 +645,6 @@ def _save(
                 "Cannot save an abstract document"
             )
 
-        if self._meta.get("auto_create_index", True):
-            self.ensure_indexes()
-
```

Comment:
Agree with the sentiment, ensure indexes every time is really annoying.
Will this have any negative externalities? I don't think so but just checking

Reply:
From what I have observed, it shouldn't as indexes are ensured on import for the dataset collection. But let me check dataset creation. Unfortunately this change impacts sample/frame collections as well

Reply:
yeah and like every other document type too right? Although the others don't have non-default (id) indexes I think so should be fine. Also I traced back the code stack and I don't really see where it's doing anything (`index_specs` not found in `fiftyone`) - but empirically I see index creation *all* the time so it must be doing something

Reply:
dataset creation definitely it happens, as well as cloning

Reply:
Yes, I'm not seeing that as well 👍